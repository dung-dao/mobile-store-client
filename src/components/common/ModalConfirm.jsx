import React from 'react';
import {Modal} from "antd";
import PropTypes from "prop-types";

const ModalConfirm = (props) => {
    return (
        <Modal
            title="Xác nhận"
            visible={props.visible}
            onOk={props.onOk}
            onCancel={props.onCancel}
            okText="Đồng ý"
            cancelText="Hủy"
        >
            {`Bạn có chắc chắn muốn ${props.actionName}?`}
        </Modal>
    );
};

ModalConfirm.propTypes = {
    visible: PropTypes.bool.isRequired,
    onOk: PropTypes.func.isRequired,
    onCancel: PropTypes.func.isRequired,
    actionName: PropTypes.string.isRequired
}

export default ModalConfirm;
