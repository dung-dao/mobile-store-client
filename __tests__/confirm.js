import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import {configure, shallow} from 'enzyme';
import ModalConfirm from "../src/components/common/ModalConfirm";

configure({adapter: new Adapter()});

configure({adapter: new Adapter()});
describe('Loading Render', () => {
    it('renders without crashing', () => {
        const tree = shallow(<ModalConfirm visible={true} onOk={() => {
            console.log('ok')
        }} onCancel={() => {
            console.log('cancel')
        }} actionName="xÃ³a"/>);
        expect(tree).toMatchSnapshot();
    });
})
