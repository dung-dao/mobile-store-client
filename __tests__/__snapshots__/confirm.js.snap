// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Render renders without crashing 1`] = `
<Modal
  cancelText="Hủy"
  confirmLoading={false}
  maskTransitionName="fade"
  okText="Đồng ý"
  okType="primary"
  onCancel={[Function]}
  onOk={[Function]}
  title="Xác nhận"
  transitionName="zoom"
  visible={true}
  width={520}
>
  Bạn có chắc chắn muốn xóa?
</Modal>
`;
